src/models/test/StyleSheetManager.test.js

```diff
@@ -86,6 +100,69 @@ describe('StyleSheetManager', () => {
     )
   })
 
+  // https://github.com/styled-components/styled-components/issues/1634
+  it('should flush styles to an iframe when the iframe is re-rendered', async () => {
+    const Title = styled.h1`
+      color: palevioletred;
+    `
+
+    // Injects the stylesheet into the document available via context
+    const SheetInjector = ({ children }, { document }) => (
+      <StyleSheetManager target={document.head}>{children}</StyleSheetManager>
+    )
+    SheetInjector.contextTypes = {
+      document: PropTypes.any,
+    }
+
+    class Child extends React.Component {
+      static contextTypes = {
+        document: PropTypes.any,
+      }
+
+      componentDidMount() {
+        // $FlowFixMe
+        const styles = this.context.document.querySelector('style').textContent
+        expect(styles.includes(`palevioletred`)).toEqual(true)
+        this.props.resolve()
+      }
+      render() {
+        return <Title />
+      }
+    }
+
+    const Structure = ({ resolve }) => (
+      <Frame>
+        <SheetInjector>
```

Comment:
So you're expecting this to run with the styles in the main window?

I'm not sure whether I can get this to work without understanding what `react-frame-component` does precisely. I probably won't have time soon to look at it, but I'll try.

Generally, are you trying to write `react-frame-component` for styled-components?

Reply:
So react-frame-component is a common library to use to put iframes into a React tree. I want to style things inside of the iframe, and to make that work you need to use a StyleSheetManager and target the iframe's head. The end-result should be that the iframe head has the styles on each render.